elide-core/src/main/java/com/yahoo/elide/security/PermissionManager.java

```diff
@@ -0,0 +1,659 @@
+        public int hashCode() {
+            return new HashCodeBuilder(3, 37)
+                    .append(resource)
+                    .append(annotation)
+                    .toHashCode();
+        }
+
+        public boolean equals(CheckIdentifier other) {
+            if (other != null || other.resource == null || other.annotation == null) {
+                return false;
+            }
+            return resource.equals(other.resource) && annotation.equals(other.annotation);
+        }
+    }
+
+    /** Security strategy */
+
+    /**
+     * Interface to encode field-aware check result response.
+     */
+    private interface Strategy {
+
+        /**
+         * Execute operation checks.
+         */
+        void executeOperationChecks();
+
+        /**
+         * Execute commit checks.
+         */
+        void executeCommitChecks();
+
+        /**
+         * Whether or not the strategy expects to execute commit checks.
+         *
+         * @return True if commit checks should be run, false otherwise.
+         */
+        boolean hasCommitChecks();
+
+        /**
+         * Status of the check state.
+         */
+        enum CheckStatus {
+            NO_CHECKS_RUN,
+            CHECKS_SUCCEEDED,
+            CHECKS_FAILED
+        }
+    }
+
+    /**
+     * Strategy for checking a specific field.
+     */
+    private final class SpecificFieldStrategy implements Strategy {
+        private CheckStatus previousStatus;
+
+        private final PersistentResource resource;
+        private final ChangeSpec changeSpec;
+
+        private Class<? extends Check>[] classOpChecks;
+        private Class<? extends Check>[] classComChecks;
+        private Class<? extends Check>[] fieldOpChecks;
+        private Class<? extends Check>[] fieldComChecks;
+        private CheckMode classCheckMode;
+        private CheckMode fieldCheckMode;
+
+        private final boolean containsDeferredChecks;
+        private final boolean isUpdate;
+
+        public <A extends Annotation> SpecificFieldStrategy(PersistentResource resource,
+                                                            String field,
+                                                            Class<A> annotationClass,
+                                                            ChangeSpec changeSpec) {
+            this.resource = resource;
+            this.previousStatus = CheckStatus.NO_CHECKS_RUN;
+            this.changeSpec = changeSpec;
+            if (field == null) {
+                log.error("Problem occurred trying to initialize Specific Field Strategy: null field.");
+                throw new IllegalStateException();
+            }
+            populateChecks(field, annotationClass);
+            containsDeferredChecks = containsCommitChecks(classComChecks, fieldComChecks, annotationClass);
+            isUpdate = UpdatePermission.class.isAssignableFrom(annotationClass);
+        }
+
+        @Override
+        public void executeOperationChecks() {
```

Comment:
Since the strategy doesn't change whether or not these are operation or commit checks, I don't think the strategy should differentiate this functions.   I think it should have a single method called execute that returns a boolean.  The calling code can decide whether to continue or not for operation checks based on whether deferred checks exist and whether the status of execute was pass/fail.